src/hazelweb/model/UndoHistory.re

```diff
@@ -464,195 +464,195 @@ let get_new_action_group =
       ~action: Action.t,
     )
     : option(UndoHistoryCore.action_group) =>
-  if (UndoHistoryCore.is_move_action(new_cursor_term_info)) {
-    None;
-        /* It's a caret movement */
-  } else {
-    switch (action) {
-    | Delete =>
-      delete(~prev_group, ~new_cardstacks_before, ~new_cursor_term_info)
-    | Backspace =>
-      backspace(~prev_group, ~new_cardstacks_before, ~new_cursor_term_info)
-    | Construct(shape) =>
-      switch (shape) {
-      | SLine =>
+  switch (action) {
+  | Delete =>
+    delete(~prev_group, ~new_cardstacks_before, ~new_cursor_term_info)
+  | Backspace =>
+    backspace(~prev_group, ~new_cardstacks_before, ~new_cursor_term_info)
+  | Construct(shape) =>
+    switch (shape) {
+    | SLine =>
+      switch (
+        CursorInfo_Exp.get_outer_zrules(new_cursor_term_info.zexp_before)
+      ) {
+      | None => Some(ConstructEdit(shape))
+      | Some(zrules_before) =>
         switch (
-          CursorInfo_Exp.get_outer_zrules(new_cursor_term_info.zexp_before)
+          CursorInfo_Exp.get_outer_zrules(new_cursor_term_info.zexp_after)
```

Comment:
@hellyab was there a bug here (unrelated to the PR)?

Reply:
@cyrus- it has always been `zexp_after`, I think git is comparing it to another line.